src/components/EditBar.vue

```diff
@@ -0,0 +1,55 @@
+<template>
+  <div class="edit-bar">
+    <span v-for="col in columns"
+      :data-column="col.name"
+      :key="col.name"
+    >
+      <GenericInput
+        :col="col"
+        :row="row"
+        :onChange="updateRow.bind(null, row.id, col.name)"
```

Comment:
Why do we use `bind()` here?

Reply:
Here, we're using bind to create a new function that acts like `(val) => updateRow(row.id, col.name)`. That is, it's first two arguments are filled in already.

We're passing `null` as the first parameter because `updateRow` doesn't care about a `this` context.